From f392d7b2cc503e183adaf33181079ab1292b3143 Mon Sep 17 00:00:00 2001
From: Le Philousophe <lephilousophe@users.noreply.github.com>
Date: Sat, 11 Mar 2023 14:16:49 +0100
Subject: [PATCH 2/3] Use CC to determine target platform

This is simpler than specifying the target.
In this case, don't check link as it may not work on some platforms
---
 build/make/configure.sh | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/build/make/configure.sh b/build/make/configure.sh
index 9c9fd54..246ec61 100644
--- a/build/make/configure.sh
+++ b/build/make/configure.sh
@@ -788,7 +788,10 @@ check_xcode_minimum_version() {
 
 process_common_toolchain() {
   if [ -z "$toolchain" ]; then
-    gcctarget="${CHOST:-$(gcc -dumpmachine 2> /dev/null)}"
+    if [ -n "$CC" ]; then
+      check_ld() { true; }
+    fi
+    gcctarget="${CHOST:-$($CC -dumpmachine 2> /dev/null)}"
     # detect tgt_isa
     case "$gcctarget" in
       aarch64*)
-- 
2.49.1

